duo_client/paginator.py

```diff
@@ -0,0 +1,198 @@
+from __future__ import absolute_import, print_function
+from time import sleep
+import six
+from .exceptions import MaxRequestAttemptsReached
+
+
+def get_backoff_seconds_exp(wait_const, min_wait_seconds, max_attempts):
+    """Calculate a counter and wait time in seconds for each retried request.
+
+    :param wait_const: The constant to apply to the exponential calculation
+    :param min_wait_seconds: Minimum wait time to yield
+    :max_attemps: Maximum number of wait times to yield
+    :returns: A generator that yields a counter and a wait time in seconds
+    :rtype: generator
+    """
+
+    for attempt in range(max_attempts):
+        yield (attempt, max(wait_const**attempt, min_wait_seconds))
+
+
+class Paginator(object):
+    """Class will retry rate limited requests + chain paginated requests."""
+
+    DEFAULT_MAX_ATTEMPTS = 4
+    DEFAULT_MIN_WAIT_SECONDS = 5
+    DEFAULT_WAIT_CONST = 5
+    DEFAULT_BACKOFF_FUNC = get_backoff_seconds_exp
+    DEFAULT_ALL_PAGES = -1
+    DEFAULT_META_OFFSET_NAME = "next_offset"
+    DEFAULT_STATUSES_TO_RETRY_ON = (429,)
+
+    def __init__(
+        self,
+        duo_client,
+        max_attempts=DEFAULT_MAX_ATTEMPTS,
+        wait_const=DEFAULT_WAIT_CONST,
+        min_wait_seconds=DEFAULT_MIN_WAIT_SECONDS,
+        backoff_func=DEFAULT_BACKOFF_FUNC,
+        statues_to_retry_on=DEFAULT_STATUSES_TO_RETRY_ON
```

Comment:
s/statues/statuses

Reply:
I didn't even realize I wrote that <facepalm>.  Thanks for noticing that!